phpBB/cron.php

```diff
@@ -23,50 +22,29 @@
 $user->session_begin(false);
 $auth->acl($user->data);
 
-function output_image()
-{
-	// Output transparent gif
-	header('Cache-Control: no-cache');
-	header('Content-type: image/gif');
-	header('Content-length: 43');
-
-	echo base64_decode('R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==');
-
-	// Flush here to prevent browser from showing the page as loading while
-	// running cron.
-	flush();
-}
-
-// Thanks to various fatal errors and lack of try/finally, it is quite easy to leave
-// the cron lock locked, especially when working on cron-related code.
-//
-// Attempt to alleviate the problem by doing setup outside of the lock as much as possible.
-
 $cron_type = $request->variable('cron_type', '');
 
-// Comment this line out for debugging so the page does not return an image.
-output_image();
+$get_params_array = $request->get_super_global(\phpbb\request\request_interface::GET);
+$separator = '?';
+$get_params = '';
 
-/* @var $cron_lock \phpbb\lock\db */
-$cron_lock = $phpbb_container->get('cron.lock_db');
-if ($cron_lock->acquire())
+foreach ($get_params_array as $name => $value)
 {
-	/* @var $cron \phpbb\cron\manager */
-	$cron = $phpbb_container->get('cron.manager');
-
-	$task = $cron->find_task($cron_type);
-	if ($task)
+	if ($name === 'cron_type')
 	{
-		if ($task->is_parametrized())
-		{
-			$task->parse_parameters($request);
-		}
-		if ($task->is_ready())
-		{
-			$task->run();
-		}
+		continue;
 	}
-	$cron_lock->release();
+
+	$get_params .= $separator . $name . '=' . $value;
+	$separator = '&';
 }
 
-garbage_collection();
+/** @var \phpbb\controller\helper $controller_helper */
+$controller_helper = $phpbb_container->get('controller.helper');
+$response = new RedirectResponse(
+	$controller_helper->route('phpbb_cron_run', array(
```

Comment:
you can pass the whole parameters array array to the route() method. They will be added as query string by the UrlGenerator if required